olp-cpp-sdk-dataservice-read/src/repositories/QuadTreeIndex.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <memory>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+#include <olp/core/geo/tiling/TileKey.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+using BlobData = std::vector<unsigned char>;
+using BlobDataPtr = std::shared_ptr<BlobData>;
+
+class QuadTreeIndex {
+ public:
+  struct IndexData {
+    /// tile key in the layer tree
+    olp::geo::TileKey tile_key;
+    /// tile path can be used in MapEngine::getData
+    std::string data_handle;
+    /// catalog version this tile was last changed at
+    uint64_t version;
+    bool operator<(const IndexData& other) const {
```

Comment:
Probably we will also need equal and unequal operators.

Reply:
What the reason?